### ISSUE TITLE
Update pyasn1 in Singularity recipe

### ISSUE THREAD

**Issue by victorsndvg** [singularityhub/singularity-python#83]
Hi,

It seems that `pyasn1==0.3.4` doesn't work now.

https://github.com/singularityware/singularity-python/blob/master/Singularity.container#L28

I changed to version `0.4.1` and It allow me to create the container. 
```
pip install pyasn1==0.4.1 &> /dev/null
```
(Now I have to check if it works properly)

What do you think?

Hope it helps!

**Comment by vsoch** [comment]
I've had trouble with that one module in some many places! Could you show me enough to reproduce the error, and then where you put that fix? I would like to reproduce the error, test your fix, and then we can resolve the issue.

**Comment by victorsndvg** [comment]
Yes, of course.

I have this Dockerfile:
```
FROM conda/miniconda3
MAINTAINER Victor Sande <https://github.com/victorsndvg>
 
RUN apt-get update \
    && apt-get install -y git \
    && cd /tmp \
    && VERSION=2.4.1 \
    && conda install -y numpy scikit-learn cython pandas \
    && git clone -b v$VERSION https://www.github.com/vsoch/singularity-python \
    && cd /tmp/singularity-python \
    && pip install setuptools \
    && pip install -r requirements.txt \
    && pip install pyasn1==0.3.4 \
    && python setup.py sdist \
    && python setup.py install \
    && cd /tmp \
    && rm -rf /tmp/singularity-python \
    && apt-get autoremove -y \
    && apt-get remove --purge -y git
```
calling to:
```
$ docker build -t victorsndvg/sregistry:2.4.1 .
```
I get the following error:
```
...
Installed /usr/local/lib/python3.6/site-packages/chardet-3.0.4-py3.6.egg
error: pyasn1 0.3.4 is installed but pyasn1<0.5.0,>=0.4.1 is required by {'pyasn1-modules'}
The command '/bin/sh -c apt-get update     && apt-get install -y git     && cd /tmp     && VERSION=2.4.1     && conda install -y numpy scikit-learn cython pandas     && git clone -b v$VERSION https://www.github.com/vsoch/singularity-python     && cd /tmp/singularity-python     && pip install setuptools     && pip install -r requirements.txt     && pip install pyasn1==0.3.4     && python setup.py sdist     && python setup.py install     && cd /tmp     && rm -rf /tmp/singularity-python     && apt-get autoremove -y     && apt-get remove --purge -y git' returned a non-zero code: 1
```
If I change `pyasn1` to `0.4.1` (as it suggest) it works for me.

The same happens in [install_sregistry.sh](https://github.com/victorsndvg/ONE_ubuntu1604_install_sregistry/blob/7b40d51155f82c2c6f6ae8cfbf0529410fe9d0be/install_sregistry.sh#L62) executed in a clean Ubuntu Xenial

I'm going to try with the Singularity recipe to confirm and will update the comment

## Update:

Same error with this Singularity recipe:

https://github.com/singularityware/singularity-python/blob/master/Singularity.container

```
Installed /opt/conda/lib/python3.6/site-packages/singularity-2.4.1-py3.6.egg
Processing dependencies for singularity==2.4.1
error: pyasn1 0.3.4 is installed but pyasn1<0.5.0,>=0.4.1 is required by {'pyasn1-modules'}
ABORT: Aborting with RETVAL=255
Cleaning up...
vsande@localhos
```

hope it helps!

**Comment by vsoch** [comment]
Hugely! I just tested the above and was able to reproduce the error. The Singularity.container file is fixed in development, and I've also added [Docker images](https://hub.docker.com/r/vanessa/singularity-python/) (I haven't pushed them yet) that are tagged with 2.4.1 to hopefully be useful to others (to either use as bases for their own custom images or just as is).

This is likely one of the errors that can be resolved with separation of the different clients. That particular module is for one of the google libraries, which we don't need for sregistry or singularity-python proper.
